src/jdk.incubator.vector/share/classes/jdk/incubator/vector/Float16.java

```diff
@@ -0,0 +1,1463 @@
+     * value; returns {@code false} otherwise (for NaN and infinity
+     * arguments).
+     *
+     * @apiNote
+     * This method corresponds to the isFinite operation defined in
+     * IEEE 754.
+     *
+     * @param f16 the {@code Float16} value to be tested
+     * @return {@code true} if the argument is a finite
+     * floating-point value, {@code false} otherwise.
+     *
+     * @see Float#isFinite(float)
+     * @see Double#isFinite(double)
+     */
+    public static boolean isFinite(Float16 f16) {
+        return (float16ToRawShortBits(f16) & (short)0x0000_7FFF) <= float16ToRawShortBits(MAX_VALUE);
+     }
+
+    /**
+     * {@return the value of this {@code Float16} as a {@code byte} after
+     * a narrowing primitive conversion}
+     *
+     * @jls 5.1.3 Narrowing Primitive Conversion
+     */
+    @Override
+    public byte byteValue() {
+        return (byte)floatValue();
+    }
+
+    /**
+     * {@return a string representation of this {@code Float16}}
+     *
+     * @implSpec
+     * This method returns the result of {@code Float16.toString(this)}.
+     */
+    public String toString() {
+        return toString(this);
+    }
+
+    /**
+     * {@return the value of this {@code Float16} as a {@code short}
+     * after a narrowing primitive conversion}
+     *
+     * @jls 5.1.3 Narrowing Primitive Conversion
+     */
+    @Override
+    public short shortValue() {
+        return (short)floatValue();
+    }
+
+    /**
+     * {@return the value of this {@code Float16} as an {@code int} after
+     * a narrowing primitive conversion}
+     *
+     * @jls 5.1.3 Narrowing Primitive Conversion
+     */
+    @Override
+    public int intValue() {
+        return (int)floatValue();
+    }
+
+    /**
+     * {@return value of this {@code Float16} as a {@code long} after a
+     * narrowing primitive conversion}
+     *
+     * @jls 5.1.3 Narrowing Primitive Conversion
+     */
+    @Override
+    public long longValue() {
+        return (long)floatValue();
+    }
+
+    /**
+     * {@return the value of this {@code Float16} as a {@code float}
+     * after a widening primitive conversion}
+     *
+     * @apiNote
+     * This method corresponds to the convertFormat operation defined
+     * in IEEE 754.
+     *
+     * @jls 5.1.2 Widening Primitive Conversion
+     */
+    @Override
+    public float floatValue() {
+        return float16ToFloat(value);
+    }
+
+    /**
+     * {@return the value of this {@code Float16} as a {@code double}
+     * after a widening primitive conversion}
+     *
+     * @apiNote
+     * This method corresponds to the convertFormat operation defined
+     * in IEEE 754.
+     *
+     * @jls 5.1.2 Widening Primitive Conversion
+     */
+    @Override
+    public double doubleValue() {
+        return (double)floatValue();
+    }
+
+    // Skipping for now:
+    // public int hashCode()
+    // public static int hashCode(Float16 value)
+    // public boolean equals(Object obj)
```

Comment:
I think we should add these methods in a subsequent PR. e.g., (although i may be ignorant of some details)
```
    @Override
    public int hashCode() {
        return Float16.hashCode(value);
    }

    public static int hashCode(float value) {
        return float16ToShortBits(value); 
    }

    public static int float16ToShortBits(float value) {
        if (!isNaN(value)) {
            return float16ToRawShortBits(value);
        }
        return Float16.NaN;
    }

    public boolean equals(Object obj) {
        return (obj instanceof Float16)
               && (float16ToShortBits(((Float16)obj).value) == float16ToShortBits(value));
    }
```

Reply:
Hmm. Yes, adding at least explicit hashCode/equals methods from Object is reasonable. Back when the type was in the Valhalla repo, I wasn't sure what the recommended practice regarding Object methods vis a vis JEP 401/402.

Reply:
Pushed equals/hashCode implementations. One slightly subtle point: the bit pattern for the "canonical" NaN is not defined. At runtime, the bit pattern of Float16.NaN is used for that purpose. Also, the exact bits returned for hashing is not defined, although I think it would be reasonable to add a requirement that "all distinct Float16 values have distinct hash codes."